_source/_posts/2023-07-27-spring-graphql-react.md

```diff
@@ -0,0 +1,1794 @@
+2023-08-02T13:11:23.693-03:00  INFO 28673 --- [           main] a.s.neo4j.migrations.core.Migrations     : Applied migration 003 ("Person").
+2023-08-02T13:21:03.680-03:00  INFO 28673 --- [           main] a.s.neo4j.migrations.core.Migrations     : Applied migration 004 ("PersonCompany").
+2023-08-02T13:21:06.519-03:00  INFO 28673 --- [           main] a.s.neo4j.migrations.core.Migrations     : Applied migration 005 ("CompanyData").
+2023-08-02T13:21:06.551-03:00  INFO 28673 --- [           main] a.s.neo4j.migrations.core.Migrations     : Applied migration 006 ("Land").
+```
+
+Test the API with GraphiQL at **http://localhost:8080/graphiql**. In the query box on the left, paste the following query
+
+```graphql
+{
+    companyList(page: 20) {
+        id
+        SIC
+        name
+        status
+        category
+        companyNumber
+        countryOfOrigin
+    }
+}
+```
+
+You should see the query output in the box on the right:
+
+{% img blog/spring-graphql-react/graphiql-test.png alt:"GraphiQL example" width:"900" %}{: .center-image }
+
+
+> __NOTE:__
+> If you see a warning message in the server logs, that reads _The query used a deprecated function: id_, you can ignore it, Spring Data Neo4j still [behaves correctly](https://github.com/spring-projects/spring-data-neo4j/issues/2716).
+
+
+## Build a React client
+
+Now let's create a Single Page Application (SPA) to consume the GraphQL API with React and Next.js. The list of companies will display in a [MUI](https://mui.com/material-ui/getting-started/) [Datagrid](https://mui.com/x/react-data-grid/) component. The application will use Next.js `app` router. The `src/app` folder will contain routing files only, and the UI components and application code will be located in other folders.
+
+Install Node and in a terminal run:
+
+```shell
+npx create-next-app@13.4.19
+```
+
+Answer the questions as follows:
+
+```
+✔ What is your project named? ... react-graphql
+✔ Would you like to use TypeScript? ... Yes
+✔ Would you like to use ESLint? ... Yes
+✔ Would you like to use Tailwind CSS? ... No
+✔ Would you like to use `src/` directory? ... Yes
+✔ Would you like to use App Router? (recommended) ... Yes
+✔ Would you like to customize the default import alias? ... No
+```
+
+Then add the the MUI Datagrid dependencies and custom hooks from Vercel:
+
+```shell
+cd react-graphql && \
+  npm install @mui/x-data-grid && \
+  npm install @mui/material@5.14.5 @emotion/react @emotion/styled && \
+  npm install react-use-custom-hooks
+```
+
+Test run the application with:
+
+```shell
+npm run dev
+```
+
+Navigate to http://localhost:3000 and you should see the default Next.js page:
+
+{% img blog/spring-graphql-react/nextjs-default.png alt:"Next.js default page" width:"800" %}{: .center-image }
+
+### Create the API client
+
+Create the folder `src/services` and add the file `base.tsx` with the following code:
+
+{% raw %}
+__base.tsx__
+```tsx
+import axios from "axios";
```

Comment:
Should Axios be installed at this point? I'm getting errors from IntelliJ, and I don't see it in `package.json`.

Reply:
Added the axios dependency. I was using double quotes because the quick fix of my IDE was configured by default with double quotes. Replaced by single quotes.